src/dftbp/dftbplus/initprogram.F90

```diff
@@ -4150,16 +4150,30 @@ end subroutine initializeCharges
   ! Assign reference charge arrays, q0 and qShell0
   subroutine initReferenceCharges(species0, orb, referenceN0, nSpin, q0, qShell0, customOccAtoms,&
       & customOccFillings)
+
+    !> Species of central cell atoms
     integer, intent(in) :: species0(:)
+
+    !> Atomic orbital data
     type(TOrbitals), intent(in) :: orb
+
+    !> Reference neutral atom charges for each species
     real(dp), intent(in) :: referenceN0(:,:)
+
+    !> Number of spin channels
     integer, intent(in) :: nSpin
-    real(dp), allocatable, intent(out) :: q0(:,:,:), qShell0(:,:)
+
+    !> Reference 'neutral' charges for each atom orbital
+    real(dp), allocatable, intent(out) :: q0(:,:,:)
+
+    !> Shell resolved neutral reference
+    real(dp), allocatable, intent(out) :: qShell0(:,:)
+
     type(TWrappedInt1), optional, intent(in) :: customOccAtoms(:)
+
     real(dp), optional, intent(in) :: customOccFillings(:,:)
```

Comment:
It is probably worth documenting these optional arguments as well.

Reply:
Done